src/Core/Services/Implementations/CipherService.cs

```diff
@@ -664,6 +664,90 @@ public async Task DeleteFolderAsync(Folder folder)
             await _pushService.PushSyncVaultAsync(importingUserId);
         }
 
+        public async Task SoftDeleteAsync(Cipher cipher, Guid deletingUserId, bool orgAdmin = false)
+        {
+            if (!orgAdmin && !(await UserCanEditAsync(cipher, deletingUserId)))
+            {
+                throw new BadRequestException("You do not have permissions to soft delete this.");
+            }
+
+            if (cipher.DeletedDate.HasValue)
+            {
+                // Already soft-deleted, we can safely ignore this
+                return;
+            }
+
+            cipher.DeletedDate = DateTime.UtcNow;
+            cipher.RevisionDate = DateTime.UtcNow;
```

Comment:
I'd suggest making these exactly the same. Could me hundredths of a ms difference here.

```suggestion
            cipher.DeletedDate = cipher.RevisionDate = DateTime.UtcNow;
```

Reply:
Made suggested change, agree